feat: handle drd fields for decisions

kristinkomschow opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->
- Add decision requirements name and version to the decision index
- Add cache for decision requirements since decision records do not contain the associated drd version and name
- Return decision requirements name and version in V2 API responses and add filter and sorting options
- Updated Camunda Client requests and responses to use the new fields

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)).

## Related issues

closes https://github.com/camunda/camunda/issues/20767

kristinkomschow committed:
style: fixed typo

kristinkomschow committed:
feat: exporter adds drd data to decision entities

- added cache for decision requirements to provide drd data for decision export
- added drd data to decision definitions

relates to https://github.com/camunda/camunda/issues/20767

kristinkomschow committed:
test: added test coverage for decision requirements cache

kristinkomschow committed:
feat: include new Decision fields in API requests and responses

- added decisionRequirementsName and decisionRequirementsVersion to API entities
- added those fields also as filter and sort options

relates to https://github.com/camunda/camunda/issues/20767

kristinkomschow committed:
feat: added new Decision fields to rdbms

- added decisionRequirementsName and decisionRequirementsVersion fields
for Decisions to RDBMs implementation

EuroLew reviewed (commented):
Reviewed the modules `webapps-schema` and everything in `zeebe` as that's data layer stuff.

Personally I'm not a fan of the repeated code for caches now that we have so many I would like to see some sort of cache registry like so 

```
public class ExporterCaches {
  private final ExporterEntityCacheImpl<BatchOperation> batchOperationCache;
  private final ExporterEntityCacheImpl<Process> processCache;
  private final ExporterEntityCacheImpl<DecisionRequirements> decisionRequirementsCache;
  
  public ExporterCaches(
      CacheConfiguration configuration,
      EntityCacheProvider entityCacheProvider,
      IndexDescriptors indexDescriptors,
      MeterRegistry meterRegistry) {
    
    this.batchOperationCache = createCache("batchOperation",
        BatchOperationTemplate.class,
        configuration.getBatchOperationCache(),
        entityCacheProvider::getBatchOperationCacheLoader,
        indexDescriptors, meterRegistry);
        
    this.processCache = createCache("process",
        ProcessIndex.class,
        configuration.getProcessCache(),
        entityCacheProvider::getProcessCacheLoader,
        indexDescriptors, meterRegistry);
        
    this.decisionRequirementsCache = createCache("decisionRequirements",
        DecisionRequirementsIndex.class,
        configuration.getDecisionRequirementsCache(),
        entityCacheProvider::getDecisionRequirementsCacheLoader,
        indexDescriptors, meterRegistry);
  }
  
  public ExporterEntityCacheImpl<BatchOperation> batchOperation() {
    return batchOperationCache;
  }
  
  public ExporterEntityCacheImpl<Process> process() {
    return processCache;
  }
  
  public ExporterEntityCacheImpl<DecisionRequirements> decisionRequirements() {
    return decisionRequirementsCache;
  }
  
  private <T> ExporterEntityCacheImpl<T> createCache(...) {
    // Same as above
  }
}
```

And then usage would be just passing the cache through

```
public class SomeService {
  private final ExporterEntityCacheImpl<BatchOperation> cache;
  
  public SomeService(ExporterCaches caches) {
    this.cache = caches.batchOperation();
  }
}

```
- you get a single dep to pass around (the cache)
- type safety
- adding new caches is easy just a new field and getter
- you can test with a mock registry.

This is just a suggestion not a must have

megglos reviewed (changes requested):
API spec and REST server changes look good to me.

What I'm missing are the camunda-client-java changes, given this closes the linked issue I assume you didn't plan for that yet? Would prefer to have them in this PR but a follow-up PR is also fine. Requesting changes to align on that.

megglos reviewed (approved):
client LGTM! 👍

kristinkomschow committed:
feat: added decisionRequirementsName and decisionRequirementsVersion to Decsion index

relates to https://github.com/camunda/camunda/issues/20767

megglos reviewed (approved):
Approving client changes from CamundaEx.

There are some schema related test failures to check though.

EuroLew commented on webapps-schema/src/main/java/io/camunda/webapps/schema/entities/dmn/definition/DecisionDefinitionEntity.java:
```diff
@@ -22,6 +23,13 @@ public class DecisionDefinitionEntity
+  private String decisionRequirementsName;
+
+  @SinceVersion("8.9.0")
+  private int decisionRequirementsVersion;
```
same here `SinceVersion` requiresDefault should be false

EuroLew reviewed (approved):
just some stuff to get the `EntityTest` to pass otherwise LGTM

kristinkomschow commented:
Fixed the remaining `@Since` issue. Let's see what the CI says.

kristinkomschow commented:
Rebased on main now that the fix for the ITs is merged

kristinkomschow commented:
Some more (unrelated) issues have been discovered for the BackupRestore tests (slack discussion [here](https://camunda.slack.com/archives/C08MRKHJ0CD/p1764604181512329)). It looks like they've been disabled for now so I rebased again.

kristinkomschow commented:
The Backup Restore ITs were still failing due to the Tasklist container creating indices with the wrong schema. It'll be fixed here: https://github.com/camunda/camunda/pull/41978
I'll rebase again after and hopefully everything finally passes then 🤞

kristinkomschow committed:
feat: add new decision fields to camunda client

- aligned decision definition variable naming with API definition naming
- added decision requirements name and version to search results, filter and sorting
- fixed tests after decoupling zeebe decisions from search response
- added decision requirement field sort and filter tests
- refactored some assertions to make them more readable

relates to https://github.com/camunda/camunda/issues/20767
